### ISSUE TITLE
[Accessibility] Calendar - Narrator does not read out day of the week, row and column numbers 

### ISSUE THREAD

**Issue by abharida** [microsoft/fluentui#16189]
### Environment Information

- **Package version(s)**: "office-ui-fabric-react": "7.121.7",
- **Browser and OS versions**: Microsoft Edge Version 87.0.664.57 (Official build) (64-bit) Windows 10 19042.685

### Describe the issue:

While using the Calendar component, Narrator does not call out the day of the week or the column or row numbers. This information is crucial especially for people with different start of the week. For reference, in the Calendar example that was available on Mozilla developers page (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Grid_Role), narrator was reading out the day of the week.

### Please provide a reproduction of the issue in a codepen:
This is reproducible in the components displayed on https://developer.microsoft.com/en-us/fluentui#/controls/web/calendar. I specifically tested this on 'Calendar with overlaid month picker launched from a button' variant.

#### Actual behavior:

Narrator does not call out the day of the week or column and row numbers.

#### Expected behavior:

Narrator calls out the day of the week or column and row numbers.

### Documentation describing expected behavior

Calendar example in (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Grid_Role)

**Comment by gouttierre** [comment]
@abharida, Thank you for reaching out to us!

Are you still experiencing these problems? As I tried with later versions, i.e. after "7.121.7" and the issue didn't repo.

**Comment by abharida** [comment]
@gouttierre I'm still having some issues with the version hosted in https://developer.microsoft.com/en-us/fluentui#/controls/web/calendar. I'm hoping that would be the latest stable version.

Repro steps

1. Open the webpage: https://developer.microsoft.com/en-us/fluentui#/controls/web/calendar
2. Go to the example titled "Calendar with overlaid month picker launched from a button" and hit that button.
3. The Calendar opens up with focus on the current date (say 8th March 2021).
4. Use the right arrow key to navigate to the next date (say 9th March 2021)
5. Narrator just reads out "March 9th 2021 button". It does not read out "Tuesday" or the row and column numbers

Test Platform

OS: Windows 10 Enterprise (Build: 19042.804)
Browser: Microsoft Edge Version 89.0.774.45 (Official build) (64-bit)
Screen Reader: Windows 10 Narrator

**Comment by pragya91** [comment]
I can reproduce this issue partially. Narrator does not announce day of the week, like Monday, Tuesday etc..
